src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.cpp

```diff
@@ -93,13 +93,69 @@ ElasticWaveEquationSEM::ElasticWaveEquationSEM( const std::string & name,
     setSizedFromParent( 0 ).
     setDescription( "Displacement value at each receiver for each timestep (z-component)" );
 
+  registerWrapper( viewKeyStruct::sourceForceString(), &m_sourceForce ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setSizedFromParent( 0 ).
+    setApplyDefaultValue( { 0.0, 0.0, 0.0 } ).
+    setDescription( "Force of the source: 3 real values for a vector source, and 6 real values for a tensor source (in Voigt notation)."
+                    "The default value is { 0, 0, 0 } (no net force)." );
+
+  registerWrapper( viewKeyStruct::sourceMomentString(), &m_sourceMoment ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setSizedFromParent( 0 ).
+    setApplyDefaultValue( { 1.0, 1.0, 1.0, 0.0, 0.0, 0.0 } ).
+    setDescription( "Moment of the source: 6 real values describing a symmetric tensor in Voigt notation."
+                    "The default value is { 1, 1, 1, 0, 0, 0 } (diagonal moment, corresponding to a pure explosion)." );
 }
 
 ElasticWaveEquationSEM::~ElasticWaveEquationSEM()
 {
   // TODO Auto-generated destructor stub
 }
 
+localIndex ElasticWaveEquationSEM::getNumNodesPerElem()
```

Comment:
Can you just move this one and `AcousticWaveEquationSEM::getNumNodesPerElem` to `WaveSolverBase` to avoid code duplication?

Also, I think the goal is to call this function in `postProcessInput` (as you do in `AcousticWaveEquationSEM`), but it seems the `numNodesPerElem` is still hard-coded to 8 there, unless I am missing something.

Reply:
You are right for both things:
- Indeed I forgot an hardcoded number, i'll fix it.
- The movind into WaveSilverBase of this function is the last part of PR #2175 which refactor several things in particular this function. I prefer to do it inside the refactor as here we only want to touch files related tio the elastic stuff.